packages/opentelemetry-instrumentation/src/instrumentation.ts

```diff
@@ -40,7 +40,7 @@ export abstract class InstrumentationAbstract<T = any>
   constructor(
     public readonly instrumentationName: string,
     public readonly instrumentationVersion: string,
-    config: types.InstrumentationConfig = {}
+    config: Partial<Config> = {},
```

Comment:
Why `Partial<Config>` here and at several other places? This allows to pass incomplete configurations. If there are really mandatory options defined in a config type defaulting to `{}` would be wrong.

Maybe we should remove the defaulting in base class and require the concrete classes to define their defaults?